openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -1580,6 +1580,10 @@ def from_smiles(
         elif hydrogens_are_explicit:
             for atom_idx in range(rdmol.GetNumAtoms()):
                 atom = rdmol.GetAtomWithIdx(atom_idx)
+                # Issue #1697: GetTotalNumHs() counts the number of hydrogens that are
+                # not present in the graph, so explicit hydrogens are not counted.
+                # For some reason this isn't the same as the number of Implicit Hydrogens
+                # Likely for cases where only same hydrogens are defined.
                 if atom.GetNumImplicitHs() != 0:
```

Comment:
(not blocking) It may be worth considering whether this first if is necessary at all. There's no harm in keeping them both, even if they're redundant (in fact this is such a highly-trafficked code path that I'd need a decent amount of convincing to remove it!). But I wonder if we can prove that there are cases that this one would catch that the second wouldn't. This would really help my understanding of the whole RDKit-implicit-hydrogens domain.